packages/sync/src/modules/Module.php

```diff
@@ -373,4 +384,67 @@ public function get_objects_by_id( $object_type, $ids ) {
 
 		return $objects;
 	}
+
+	/**
+	 * Gets a list of minimum and maximum object ids for each batch based on the given batch size.
+	 *
+	 * @access public
+	 *
+	 * @param int $batch_size The batch size for objects.
+	 *
+	 * @return array An array of min and max ids for each batch.
+	 */
+	public function get_min_max_object_ids_for_batches( $batch_size ) {
+		global $wpdb;
+		$results     = array();
+		$table       = $wpdb->{$this->name()};
+		$current_max = 0;
+		$current_min = 1;
+		$id_field    = $this->id_field();
+		$total       = $this->get_min_max_object_id( false, $id_field, $table, false );
+		while ( $total->max > $current_max ) {
+			$where  = "WHERE $id_field > $current_max";
+			$result = $this->get_min_max_object_id( $batch_size, $id_field, $table, $where );
+			if ( empty( $result->min ) && empty( $result->max ) ) {
+				$current_max = (int) $total->max;
+				$result      = (object) array(
+					'min' => $current_min,
+					'max' => $current_max,
+				);
+			} else {
```

Comment:
Could only one of `$result->min` or `$result->max` be empty?

Reply:
AFAIK, no.